frontend/src/components/RegistrationView/index.tsx

```diff
@@ -2,39 +2,77 @@ import React, { useState, useEffect } from 'react';
 import RegistrationForm from './Form';
 import { Wrapper, Header, Confirmation, SuccessIcon, Error, Button, LogoutButton, LogoutLink } from './StyledComponents';
 import { get, post } from '../../apis';
+import { RouteComponentProps } from 'react-router-dom';
+import { isEmpty } from 'lodash';
 
-const RegistrationView: React.FC = (props) => {
-    const  [submitted, setSubmitted] = useState(false);
+
+const RegistrationView: React.FC<RouteComponentProps> = (props) => {
+    const [submitted, setSubmitted] = useState(false);
     const [clubs, setClubs] = useState([]);
     const [error, setError] = useState(false);
     const [auth, setAuth] = useState(false);
+    const [securityContext, setSecurityContext] = useState({})
+
+    const queryToSecurityContext = () => {
+        const query = new URLSearchParams(props.location.search);
+        const sc_encoded = query.get('sc');
+        if (sc_encoded) {
+            let b64str = sc_encoded.replace(/-/g, '+').replace(/_/g, '/');
+            const pad = sc_encoded.length % 4;
+            if (pad) {
+                if (pad === 1) {
+                    // TODO: this is an error, throw exception or something?
+                    console.log('base64url string is the wrong length');
+                }
+                b64str += new Array(5-pad).join('=');
+            }
+            const sc = atob(b64str);
+            sessionStorage.setItem('sc', sc);
+        }
+    }
+
+    const getSecurityContext = () => {
+        const sc = sessionStorage.getItem('sc');
+        return sc ? JSON.parse(sc) : {};
+    }
 
     useEffect(()=> {
-        post('/auth/validate-signature', {}) // TODO pass the security context, not empty object
+        queryToSecurityContext();
+        const sc = getSecurityContext();
+        setSecurityContext(sc); // <- TODO what does this do?
```

Comment:
I dont think you need to specifically set security context to react state, since it's stored to sessionStorage anyway?